src/uct/ib/base/ib_iface.c

```diff
@@ -381,14 +448,15 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
 
     uct_ib_address_unpack(ib_addr, &lid, &gid);
 
-    if (!is_local_eth && (ib_addr->flags & UCT_IB_ADDRESS_FLAG_LINK_LAYER_IB)) {
+    if (!is_local_eth && (ib_addr->flags & UCT_IB_FLAG_LINK_LAYER_IB)) {
         /* same subnet prefix */
         return gid.global.subnet_prefix == iface->gid.global.subnet_prefix;
-    } else if (is_local_eth && (ib_addr->flags & UCT_IB_ADDRESS_FLAG_LINK_LAYER_ETH)) {
-        /* there shouldn't be a lid and the gid flag should be on */
-        ucs_assert(ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID);
-        ucs_assert(!(ib_addr->flags & UCT_IB_ADDRESS_FLAG_LID));
-        return 1;
+    } else if (is_local_eth && (ib_addr->flags & UCT_IB_FLAG_LINK_LAYER_ETH)) {
+        /* there shouldn't be a lid and the UCT_IB_FLAG_LINK_LAYER_IB flag should
+         * be on. If reachable, the remote and local RoCE versions and address
+         * families have to be the same */
+        ucs_assert(!(ib_addr->flags & UCT_IB_FLAG_LINK_LAYER_IB));
```

Comment:
need to avoid comparing roce version in the address to iface->roce_ver for rdmacm flow